### PR TITLE
fix typehints

### PR DESCRIPTION
**Pull request by JaskRendix**
progressing #1211 updated mypy, always --strict
from
```
Found 260 errors in 48 files (checked 291 source files)
-- strict Found 305 errors in 57 files (checked 291 source files)
```
to
```
Found 214 errors in 46 files (checked 291 source files)
--strict Found 248 errors in 53 files (checked 291 source files)
```

### CODE REVIEW

File: tuxemon/map.py [hunk 1]
```diff
@@ -464,5 +464,6 @@ def reload_tiles(self) -> None:
             image_loader=scaled_image_loader,
             pixelalpha=True,
         )
+        assert self.renderer
```
Comment (kerizane) [inline tuxemon/map.py:467]: Indentation looks wrong

File: tuxemon/map_loader.py [hunk 2]
```diff
@@ -373,30 +373,37 @@ def load_event(
...
-                condition = MapCondition(
-                    cond_type, args, x, y, w, h, operator, key
-                )
-                conds.append(condition)
-            elif key.startswith("act"):
-                act_type, args = parse_action_string(value)
-                action = MapAction(act_type, args, key)
-                acts.append(action)
+            assert key
+            if isinstance(key, str):
```
Comment (kerizane) [inline tuxemon/map_loader.py]: Let's do

```
if not is instance(key, str):
    continue
```

Comment (kerizane) [reply tuxemon/map_loader.py]: To avoid increasing the indentation

**Review by kerizane — CHANGES_REQUESTED** [review]
Comment

File: tuxemon/map_loader.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/map_loader.py]: it makes sense

**Review by kerizane — APPROVED** [review]

**Pull request merged**